jormungandr/src/network/bootstrap.rs

```diff
@@ -83,7 +92,18 @@ pub fn bootstrap_from_peer(
                 .map_err(Error::PullRequestFailed)
                 .and_then(|stream| bootstrap_from_stream(blockchain, tip, stream, logger.clone()))
         })
-        .and_then(move |tip| branch.update_ref(tip).map_err(|_| unreachable!()));
+        .and_then(move |tip| {
+            branch
+                .update_ref(Arc::clone(&tip))
+                .map_err(|_| unreachable!())
+                .map(|_prev_tip| tip)
+        })
+        .and_then(move |tip| {
+            storage
+                .put_tag(MAIN_BRANCH_TAG.to_owned(), tip.hash())
```

Comment:
`put_tag` could be extended to accept an `Into<String>` as a tag name.

Reply:
I've created an issue to update the API #791